src/core/utils/KeyMapping.ts

```diff
@@ -0,0 +1,62 @@
+import { defaultKeyMap } from './DefaultKeyMap';
+
+export interface KeyModifiers {
+    ctrlKey?: boolean;
+    altKey?: boolean;
+    shiftKey?: boolean;
+}
+export interface Keypress {
+    key: string;
+    modifiers: KeyModifiers;
+}
```

Comment:
To echo the things said before, I would remove the `modifiers` intermediary and just have the modifiers booleans directly on this interface.